FOCUS #165: Pricing Quantity (#225)
FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec: specification/metrics/quantityinpricingunit.md

> # Quantity in Pricing Unit

Reviewer: As we've seen columns mature and close on a common prefixed grouping pattern, I would really like to see this and `QuantityInUsageUnit` be called `PricingQuantity` and `UsageQuantity` so they sit next to their respective unit columns. These will be very out-of-place if not sitting next to the column that explains what the number means.

Author: @ijurica  do we need to vote on this change or is it something you have already agreed with the maintainers?

Author: Will be amended following results of poll....

Author: As discussed in WG session ... moving to PricingQuantity for column name

> The Quantity in Pricing Unit represents the volume of a given resource and/or service used or purchased based on the [PricingUnit](#pricingunit). This quantity value is used to calculate cost (List, Billed & Effective costs) when multiplied by the associated unit price metric.

Reviewer 2: ```suggestion
The Quantity in Pricing Unit represents the volume of a given resource and/or service used or purchased based on the [PricingUnit](#pricingunit). This quantity value is used to calculate cost (List, Billed & Effective costs) when multiplied by the associated unit price metric. The Quantity In Pricing Unit pertains to pricing and cost determination and should not be confused with the [Quantity In Usage Unit}(#quantityinusageunit), which is complementary to [Usage Unit](#usageunit) and intended for for tracking resource and service consumption.
```

> The Quantity in Pricing Unit represents the volume of a given resource and/or service used or purchased based on the [Pricing Unit](#pricingunit). This quantity value is used to calculate cost (List, Billed & Effective costs) when multiplied by the associated unit price metric. The Quantity In Pricing Unit pertains to pricing and cost determination and should not be confused with the [Quantity In Usage Unit}(#quantityinusageunit), which is complementary to [Usage Unit](#usageunit) and intended for tracking resource and service consumption.

Reviewer 2: 1.0 includes the following:

- costs: BilledCost, EffectiveCost, ListCost, NegotiatedCost (optional)
- unit prices: ListUnitPrice, NegotiatedUnitPrice (optional)

Due to this, I recommend 

- **OPTION A:** removing this sentence: `This quantity value is used to calculate cost (List, Billed & Effective costs) when multiplied by the associated unit price metric.`
- **OPTION B:** rephrasing it to something like: `This quantity value can be used to calculate the unit price (e.g., List, Billed, or Effective Unit Price) by dividing the corresponding cost metric.` 

Need just one vote to proceed with the change

Reviewer 3: 1

Reviewer 4: A

Reviewer 2: ```suggestion
The Quantity in Pricing Unit represents the volume of a given resource and/or service used or purchased based on the [Pricing Unit](#pricingunit). The Quantity In Pricing Unit pertains to pricing and cost, and should not be confused with the [Quantity In Usage Unit](#quantityinusageunit), which is complementary to [Usage Unit](#usageunit) and intended for tracking resource and service consumption.
```

> The Quantity in Pricing Unit represents the volume of a given resource and/or service used or purchased based on the [PricingUnit](#pricingunit). This quantity value is used to calculate cost (List, Billed & Effective costs) when multiplied by the associated unit price metric. The Quantity In Pricing Unit pertains to pricing and cost determination and should not be confused with the [Quantity In Usage Unit}(#quantityinusageunit), which is complementary to [Usage Unit](#usageunit) and intended for for tracking resource and service consumption.

Reviewer 2: ```suggestion
The Quantity in Pricing Unit represents the volume of a given resource and/or service used or purchased based on the [Pricing Unit](#pricingunit). This quantity value is used to calculate cost (List, Billed & Effective costs) when multiplied by the associated unit price metric. The Quantity In Pricing Unit pertains to pricing and cost determination and should not be confused with the [Quantity In Usage Unit}(#quantityinusageunit), which is complementary to [Usage Unit](#usageunit) and intended for for tracking resource and service consumption.
```

> The QuantityInPricingUnit column MUST be present in the billing data. This column MUST be a numeric value of type Decimal. The value MAY be negative in cases where [ChargeType](#chargetype) is 'Adjustment'. This column MUST NOT contain null values where unit prices are not null. When unit prices are not null, multiplying QuantityInPricingUnit by a unit price MUST produce a result equal to the corresponding cost metric.

Reviewer: Standardizing quotes...
```suggestion
The QuantityInPricingUnit column MUST be present in the billing data. This column MUST be a numeric value of type Decimal. The value MAY be negative in cases where [ChargeType](#chargetype) is "Adjustment". This column MUST NOT contain null values where unit prices are not null. When unit prices are not null, multiplying QuantityInPricingUnit by a unit price MUST produce a result equal to the corresponding cost metric.
```

> The QuantityInPricingUnit column MUST be present in the billing data. This column MUST be a numeric value of type Decimal. The value MAY be negative in cases where [ChargeType](#chargetype) is "Adjustment". This column MUST NOT contain null values where unit prices are not null. When unit prices are not null, multiplying QuantityInPricingUnit by a unit price MUST produce a result equal to the corresponding cost metric.

Reviewer 5: style "guide" (? not sure if this is more just a convention)

When referring to the column, just refer to it as its ID, without additional 'the' or 'column'.

Open to discussion on it, but have seen varying usage within these PRs.

```suggestion
QuantityInPricingUnit MUST be present in the billing data. This column MUST be a numeric value of type Decimal. The value MAY be negative in cases where [ChargeType](#chargetype) is "Adjustment". This column MUST NOT contain null values where unit prices are not null. When unit prices are not null, multiplying QuantityInPricingUnit by a unit price MUST produce a result equal to the corresponding cost metric.
```

Reviewer 6: There's a few of these that we will scrub at the end across all.

> The QuantityInPricingUnit column MUST be present in the billing data. This column MUST be a numeric value of type Decimal. The value MAY be negative in case of adjustments. This column MUST NOT contain null values where unit prices are not null. When unit prices are not null, multiplying [QuantityInPricingUnit](#quantityinpricingunit) by a unit price MUST produce a result equal to the corresponding cost metric.

Reviewer 2: ```suggestion
The QuantityInPricingUnit column MUST be present in the billing data. This column MUST be a numeric value of type Decimal. The value MAY be negative in cases where the [ChargeType](#chargetype) is 'Adjustment'. This column MUST NOT contain null values where unit prices are not null. When unit prices are not null, multiplying [QuantityInPricingUnit](#quantityinpricingunit) by a unit price MUST produce a result equal to the corresponding cost metric.
```

## Column ID

QuantityInPricingUnit

## Display name

Quantity In Pricing Unit

## Description

> The volume of a given resource and/or service used or purchased based on the [PricingUnit](#pricingunit).

Reviewer 2: ```suggestion
The volume of a given resource and/or service used or purchased based on the Pricing Unit.
```
